src/v2/trading.ts

```diff
@@ -0,0 +1,81 @@
+import { coroutine as co } from 'bluebird';
+
+export class Trading {
+  private bitgo;
+  private basecoin;
+  private wallet;
+
+  constructor(bitgo, basecoin, wallet) {
+    this.bitgo = bitgo;
+    this.basecoin = basecoin;
+    this.wallet = wallet;
+  }
+
+  /**
+   * Builds a payload authorizing a trade from this trading account. The currency and amount must be specified, as well as a list
+   * of trade counterparties.
+   * @param params
+   * @param params.currency the currency this wallet will be sending as part of the trade
+   * @param params.amount the amount of currency (in base units, such as cents, satoshis, or wei) authorized to be spent as part of the trade
+   * @param params.otherParties array of trading account IDs authorized to receive funds as part of the trade
+   * @param callback
+   * @returns unsigned trade payload for the given parameters
+   */
+  buildTradePayload(params: BuildPayloadParameters, callback?): Promise<string> {
```

Comment:
The return type of co is actually bluebird\<thing\> (if you have `import * as bluebird from 'bluebird';` at the top). There are a few extra functions on bluebird promises which get lost if you just return a vanilla promise

Reply:
Had to use the legacy `import Bluebird = require('bluebird')` to get this to work, but fixed, thx